crates/astria-sequencer/src/accounts/state_ext.rs

```diff
@@ -0,0 +1,72 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use penumbra_storage::{
+    StateRead,
+    StateWrite,
+};
+
+pub type Address = String;
+pub type Balance = u64; // might need to be larger
+pub type Nonce = u32;
+
+const ACCOUNTS_PREFIX: &str = "accounts";
+
+fn storage_key(address: &str) -> String {
+    format!("{}/{}", ACCOUNTS_PREFIX, address)
+}
+
+pub(crate) fn balance_storage_key(address: &str) -> String {
+    format!("{}/balance", storage_key(address))
+}
+
+pub(crate) fn nonce_storage_key(address: &str) -> String {
+    format!("{}/nonce", storage_key(address))
+}
+
+#[async_trait]
+pub trait StateReadExt: StateRead {
+    async fn get_account_balance(&self, address: &str) -> Result<Balance> {
+        let bytes = self
+            .get_raw(&balance_storage_key(address))
+            .await
+            .expect("storage error");
+        let Some(bytes) = bytes else {
+            // the account has not yet been initialized; return 0
+            return Ok(0);
```

Comment:
This means you cannot distingish between "uninitialized" and "zero balance". Maybe `get_account_balance -> Result<Option<Balance>>`?

Reply:
this is the behaviour I want for now (and mirrors most chains like ethereum) - all accounts technically already exist, but whether anyone knows the private key for them is unknown